app/client/src/components/shared/AddDataWidget.SearchPanel.tsx

```diff
@@ -740,59 +790,60 @@ function ResultCard({ result }: ResultCardProps) {
     if (!mapView?.map) return;
 
     setStatus('loading');
 
-    Layer.fromPortalItem({
-      portalItem: new PortalItem({
-        id: result.id,
-      }),
-    }).then((layer) => {
-      // setup the watch event to see when the layer finishes loading
-      const newWatcher = watchUtils.watch(
-        layer,
-        'loadStatus',
-        (loadStatus: string) => {
-          // set the status based on the load status
-          if (loadStatus === 'loaded') {
-            setStatus('');
-
-            // set the min/max scale for tile layers
-            if (layer.type === 'tile') {
-              const tileLayer = layer;
-              tileLayer.minScale = 0;
-              tileLayer.maxScale = 0;
-            }
-
-            if (mapView) {
-              layer.visible = true;
-
-              // make all child layers visible, if applicable
-              if (layer.layers) {
-                layer.layers.items.forEach((tempLayer) => {
-                  tempLayer.visible = true;
-                });
-              }
-              if (layer.sublayers) {
-                layer.sublayers.items.forEach((tempLayer) => {
-                  tempLayer.visible = true;
-                });
-              }
-            }
-          } else if (loadStatus === 'failed') {
-            setStatus('error');
+    const layer: Layer | __esri.TileLayer | __esri.GroupLayer =
+      await Layer.fromPortalItem({
+        portalItem: new PortalItem({
+          id: result.id,
+        }),
+      });
+    // setup the watch event to see when the layer finishes loading
+    const newWatcher = watchUtils.watch(
+      layer,
+      'loadStatus',
+      (loadStatus: string) => {
+        // set the status based on the load status
+        if (loadStatus === 'loaded') {
+          setStatus('');
+
+          // set the min/max scale for tile layers
+          if (isTileLayer(layer)) {
+            const tileLayer = layer;
+            tileLayer.minScale = 0;
+            tileLayer.maxScale = 0;
           }
-        },
-      );
 
-      setWatcher(newWatcher);
+          layer.visible = true;
 
-      // add the layer to the map
-      setWidgetLayers((currentWidgetLayers) => [...currentWidgetLayers, layer]);
-    });
-  }
+          // make all child layers visible, if applicable
+          if (isGroupLayer(layer)) {
+            layer.layers.forEach((tempLayer: Layer) => {
+              tempLayer.visible = true;
+            });
+          }
+          if ('sublayers' in layer) {
+            layer.sublayers.forEach((tempLayer: __esri.Sublayer) => {
+              tempLayer.visible = true;
+            });
+          }
+        } else if (loadStatus === 'failed') {
+          setStatus('error');
+        }
+      },
+    );
+
+    setWatcher(newWatcher);
+
+    // add the layer to the map
+    setWidgetLayers((currentWidgetLayers: WidgetLayer[]) => [
+      ...currentWidgetLayers,
+      layer,
+    ]);
+  }, [mapView, result.id, setWidgetLayers]);
```

Comment:
```suggestion
      ]);
    });
  }, [mapView, result.id, setWidgetLayers]);
```
Continuation of above.